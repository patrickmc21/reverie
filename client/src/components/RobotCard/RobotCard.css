.robot-card__list {
  display: grid;
  grid-template-columns: 15% 15% 15% 15% 15% 12.5% 5% 5%;
  grid-template-rows: 50px;
  list-style: none;
}

.robot-card__list--list-item {
  grid-column-start: span 1;
  color: #FFFFFF;
}

.edit-robot,
.delete-robot {
  border: none;
  height: 20px;
  width: 20px;
  background-color: transparent;
  background-size: cover;
  background-repeat: no-repeat;
}

.edit-robot {
  background-image: url('../../assets/edit.png');
}

.edit-robot:hover {
  background-image: url('../../assets/edit-hover.png');
}

.delete-robot {
  background-image: url('../../assets/trash.png');
}

.delete-robot:hover {
  background-image: url('../../assets/trash-hover.png');
}


