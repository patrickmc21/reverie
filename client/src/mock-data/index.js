export const mockRobots = [
  {
    id: 1,
    date_added: '5/21/2018',
    first_active: '5/21/2018',
    current_name: "R2-D2",
    height: 3.25,
    weight: 200,
    intelligence_metric: 18
  },
  {
    id: 2,
    date_added: '5/20/2018',
    first_active: '5/20/2018',
    current_name: "Bender",
    height: 6.1,
    weight: 250,
    intelligence_metric: 14
  }
];

export const newRobot = {
  date_added: '5/23/2018',
  first_active: '5/23/2018',
  current_name: "T-1000",
  height: 6.5,
  weight: 400,
  intelligence_metric: 19
};

export const updatedRobot = {
  id: 2,
  date_added: '5/23/2018',
  first_active: '5/23/2018',
  current_name: "T-1000",
  height: 6.5,
  weight: 400,
  intelligence_metric: 19
};

