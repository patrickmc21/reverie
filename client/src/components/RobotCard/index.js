import React from 'react';
import './RobotCard.css';

const RobotCard = ({robot, triggerForm, removeRobot, classType}) => {
  return (
    <article className={`robot-card ${classType}`}>
      <ul className='robot-card__list'>
        <li className='robot-card__list--list-item'>
          {robot.current_name} 
        </li>
        <li className='robot-card__list--list-item'>
          {robot.height}
        </li>
        <li className='robot-card__list--list-item'>
          {robot.weight}
        </li>
        <li className='robot-card__list--list-item'>
          {robot.intelligence_metric}
        </li>
        <li className='robot-card__list--list-item'>
          {robot.date_added.substring(0, 10)}
        </li>
        <li className='robot-card__list--list-item'>
          {robot.first_active.substring(0, 10)}
        </li>
        <li className='robot-card__list--list-item'>
          <button 
            className='edit-robot'
            onClick={() => triggerForm(robot)}></button>
        </li>
        <li className='robot-card__list--list-item'>
          <button
            className='delete-robot'
            onClick={() => removeRobot(robot.id)}></button>
        </li>
      </ul>
    </article>
  )
};

export default RobotCard;