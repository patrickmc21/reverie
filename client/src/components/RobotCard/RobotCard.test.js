import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import RobotCard from './index';
import { mockRobots } from '../../mock-data';

describe('RobotCard', () => {

  let wrapper;
  let mockRobot;
  let mockTriggerForm;
  let mockRemoveRobot;

  beforeEach(() => {
    mockRobot = mockRobots[0];
    mockTriggerForm = jest.fn();
    mockRemoveRobot = jest.fn();
    wrapper = shallow(
      <RobotCard
       robot={mockRobots[0]}
       triggerForm={mockTriggerForm}
       removeRobot={mockRemoveRobot}
       classType='even' />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call triggerForm on button click', () => {
    const expected = mockRobot;
    wrapper.find('.edit-robot').simulate('click');
    expect(mockTriggerForm).toHaveBeenCalledWith(expected);
  });

  it('should call removeRobot on button click', () => {
    const expected = mockRobot.id;
    wrapper.find('.delete-robot').simulate('click');
    expect(mockRemoveRobot).toHaveBeenCalledWith(expected);
  });
});