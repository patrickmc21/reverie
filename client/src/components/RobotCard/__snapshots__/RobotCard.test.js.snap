// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RobotCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RobotCard
    classType="even"
    removeRobot={[Function]}
    robot={
        Object {
            "current_name": "R2-D2",
            "date_added": "5/21/2018",
            "first_active": "5/21/2018",
            "height": 3.25,
            "id": 1,
            "intelligence_metric": 18,
            "weight": 200,
          }
    }
    triggerForm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="robot-card__list"
>
        <li
                className="robot-card__list--list-item"
        >
                R2-D2
        </li>
        <li
                className="robot-card__list--list-item"
        >
                3.25
        </li>
        <li
                className="robot-card__list--list-item"
        >
                200
        </li>
        <li
                className="robot-card__list--list-item"
        >
                18
        </li>
        <li
                className="robot-card__list--list-item"
        >
                5/21/2018
        </li>
        <li
                className="robot-card__list--list-item"
        >
                5/21/2018
        </li>
        <li
                className="robot-card__list--list-item"
        >
                <button
                        className="edit-robot"
                        onClick={[Function]}
                />
        </li>
        <li
                className="robot-card__list--list-item"
        >
                <button
                        className="delete-robot"
                        onClick={[Function]}
                />
        </li>
</ul>,
      "className": "robot-card even",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="robot-card__list--list-item"
>
            R2-D2
</li>,
          <li
            className="robot-card__list--list-item"
>
            3.25
</li>,
          <li
            className="robot-card__list--list-item"
>
            200
</li>,
          <li
            className="robot-card__list--list-item"
>
            18
</li>,
          <li
            className="robot-card__list--list-item"
>
            5/21/2018
</li>,
          <li
            className="robot-card__list--list-item"
>
            5/21/2018
</li>,
          <li
            className="robot-card__list--list-item"
>
            <button
                        className="edit-robot"
                        onClick={[Function]}
            />
</li>,
          <li
            className="robot-card__list--list-item"
>
            <button
                        className="delete-robot"
                        onClick={[Function]}
            />
</li>,
        ],
        "className": "robot-card__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "R2-D2",
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": "R2-D2",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 3.25,
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": 3.25,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 200,
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": 200,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 18,
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": 18,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "5/21/2018",
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": "5/21/2018",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "5/21/2018",
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": "5/21/2018",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="edit-robot"
              onClick={[Function]}
/>,
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "edit-robot",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="delete-robot"
              onClick={[Function]}
/>,
            "className": "robot-card__list--list-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "delete-robot",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="robot-card__list"
>
          <li
                    className="robot-card__list--list-item"
          >
                    R2-D2
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    3.25
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    200
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    18
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    5/21/2018
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    5/21/2018
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    <button
                              className="edit-robot"
                              onClick={[Function]}
                    />
          </li>
          <li
                    className="robot-card__list--list-item"
          >
                    <button
                              className="delete-robot"
                              onClick={[Function]}
                    />
          </li>
</ul>,
        "className": "robot-card even",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="robot-card__list--list-item"
>
              R2-D2
</li>,
            <li
              className="robot-card__list--list-item"
>
              3.25
</li>,
            <li
              className="robot-card__list--list-item"
>
              200
</li>,
            <li
              className="robot-card__list--list-item"
>
              18
</li>,
            <li
              className="robot-card__list--list-item"
>
              5/21/2018
</li>,
            <li
              className="robot-card__list--list-item"
>
              5/21/2018
</li>,
            <li
              className="robot-card__list--list-item"
>
              <button
                            className="edit-robot"
                            onClick={[Function]}
              />
</li>,
            <li
              className="robot-card__list--list-item"
>
              <button
                            className="delete-robot"
                            onClick={[Function]}
              />
</li>,
          ],
          "className": "robot-card__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "R2-D2",
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": "R2-D2",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3.25,
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": 3.25,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 200,
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": 200,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 18,
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": 18,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5/21/2018",
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": "5/21/2018",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5/21/2018",
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": "5/21/2018",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="edit-robot"
                onClick={[Function]}
/>,
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "edit-robot",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="delete-robot"
                onClick={[Function]}
/>,
              "className": "robot-card__list--list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "delete-robot",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
